src/profile-logic/process-profile.js

```diff
@@ -1014,6 +1088,10 @@ export function processProfile(
     );
   }
 
+  const visualProgress: Counter[] | null = geckoProfile.meta.visualMetrics
```

Comment:
I think the visual progress is only using the counters data structure because it's convenient to use the memory track. I'm concerned that this approach needless complicates and confuses two concepts in order to not duplicate a bit of canvas code.

Reply:
I completely agree and I've made changes to my approach. Visual progress data now has it's own shape and a new component has been created to display the charts. Furthermore, visual progress data no longer needs to be processed and is ready to consume.